x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/aggregate/Percentile.java

```diff
@@ -21,26 +20,26 @@
 import static org.elasticsearch.xpack.ql.expression.TypeResolutions.isFoldable;
 import static org.elasticsearch.xpack.ql.expression.TypeResolutions.isNumeric;
 
-public class Percentile extends NumericAggregate implements EnclosedAgg {
+public class Percentile extends HasPercentilesConfig {
 
     private final Expression percent;
 
-    public Percentile(Source source, Expression field, Expression percent) {
-        super(source, field, singletonList(percent));
+    public Percentile(Source source, Expression field, Expression percent, Expression method, Expression methodParameter) {
+        super(source, field, singletonList(percent), method, methodParameter);
         this.percent = percent;
     }
 
     @Override
     protected NodeInfo<Percentile> info() {
-        return NodeInfo.create(this, Percentile::new, field(), percent);
+        return NodeInfo.create(this, Percentile::new, field(), percent, method(), methodParameter());
     }
 
     @Override
     public Percentile replaceChildren(List<Expression> newChildren) {
         if (newChildren.size() != 2) {
             throw new IllegalArgumentException("expected [2] children but received [" + newChildren.size() + "]");
         }
-        return new Percentile(source(), newChildren.get(0), newChildren.get(1));
+        return new Percentile(source(), newChildren.get(0), newChildren.get(1), method(), methodParameter());
     }
 
     @Override
```

Comment:
I believe this method is the same for Percentile/PercentileRanks so it can be extracted into the parent.

Reply:
See https://github.com/elastic/elasticsearch/pull/65026#discussion_r526953312